### DISCUSSION TITLE
Issue with installing Biostar handbook envt bioinfo

### DISCUSSION THREAD

**Discussion by anjanaram1** [biostars/biostar-handbook-issues#287] [Q&A]
Hello,

I recently purchased the biostar handbook and am running into some tricky situations trying to install it, when creating the bioinfo environment for micromamba; i keep getting the error-

_`critical libmamba Cannot activate, prefix does not exist at: '/Users/anjanaram/micromamba/envs/bioinfo'`_

**Comment by ialbert** [comment]
This sounds like the environment was not created for some reason.
```
conda activate foo
critical libmamba Cannot activate, prefix does not exist at: '/home/ialbert/micromamba/envs/foo'
```

Have you followed the installation instructions in the current version of the book?

**Reply by anjanaram1** [reply]
How do I know if I am using the latest/current version of the book? thanks,

**Reply by ialbert** [reply]
The current version of the book is online and you can both directly read it or download it from there.

**Reply by anjanaram1** [reply]
In that case, yes, I am following the most current version. 

It is unable to create the bioinfo envt

**Reply by ialbert** [reply]
what does it mean "unable to create"? 

When indicating a problem you have to be specific as to what the error message is.

**Reply by anjanaram1** [reply]
When I run the command from the handbook- 

`curl http://data.biostarhandbook.com/install.sh | bash`

I get the error msg-

_tar: Unrecognized archive format
tar: bin/micromamba: Not found in archive
tar: Error exit delayed from previous errors.__

However, micromamba is installed.

**Reply by ialbert** [reply]
there is a step-by-step installation instruction page that you can follow, where each step is shown separately

please follow that and report which step fails

**Reply by anjanaram1** [reply]
Yes, that is what I am following and the above step in the process failed.

**Reply by ialbert** [reply]
show the command that fails in the step by step instruction, both the command and the error message

what you should above is not the step-by-step stage but the main install

**Reply by anjanaram1** [reply]
During the set up, xcode installs successfully but when I try to install the command line by executing -

`xcode-select --install`

i get the error-

_xcode-select: error: command line tools are already installed, use "Software Update" to install updates_

**Reply by ialbert** [reply]
this is not an "error"  - though the language of the message is confusing, I would agree

note how it states that *the tools are already installed.*  - basically it says this step has already been completed no need to run it again

you can move on to the next step there

**Reply by anjanaram1** [reply]
Yes-so the next part where I get the error is given above in the original post i.e

_critical libmamba Cannot activate, prefix does not exist at: '/Users/anjanaram/micromamba/envs/bioinfo'_

**Reply by ialbert** [reply]
did the environment creation work?

what happened when you created the environment as described in steps 4, 5 and 6. Did all those steps complete with no error?

**Reply by anjanaram1** [reply]
in Step 6: Create the bioinfo environment when i do - 

`$ conda create -n bioinfo -y python=2.7`

is where I encounter my first error - 

_dyld: lazy symbol binding failed: Symbol not found: _utimensat
  Referenced from: /Users/anjanaram/./bin/micromamba
  Expected in: /usr/lib/libSystem.B.dylib_

**Reply by ialbert** [reply]
you are writing python=2.7 above is that what you used? 

Because as far as I know the book does not use python=2.7 anywhere

**Reply by anjanaram1** [reply]
This is the version of python I have on my computer hence used it

**Reply by ialbert** [reply]
you need to follow the instructions as provided in the book, don't try and substitute commands unless you fully understand why something is written a specific way

when you create an environment, you get  a new version of python within that environment that is independent of the system python. That is the whole point of environments.

**Reply by anjanaram1** [reply]
got it, thanks.

**Reply by anjanaram1** [reply]
I am following the step by step instructions as indicated. Ive created the bin directory. Everything is smooth until this command step-

$micromamba create -n bioinfo -y python=3.8

Whereupon I get the error

$ dyld: lazy symbol binding failed: Symbol not found: _utimensat
  Referenced from: /Users/anjanaram/./bin/micromamba
  Expected in: /usr/lib/libSystem.B.dylib

Conda seems to be installing ok. The issue appears to be with the bioconda package.

Please note-I do have Anaconda installed already not sure if that changes anything.
